hoomd/hpmc/integrate.py

```diff
@@ -369,1046 +374,1249 @@ def type_shapes(self):
+                                            vertices=list,
+                                            sweep_radius=0.0,
+                                            ignore_statistics=False,
+                                            len_keys=1)
+                                        )
+        self._add_typeparam(typeparam_shape)
 
-    def get_type_shapes(self):
+    @Loggable.log(flag='multi')
+    def type_shapes(self):
         """Get all the types of shapes in the current simulation.
 
         Example:
-            >>> mc.get_type_shapes()
+            >>> mc.type_shapes()
             [{'type': 'ConvexPolyhedron', 'rounding_radius': 0.1,
               'vertices': [[0.5, 0.5, 0.5], [0.5, -0.5, -0.5],
                            [-0.5, 0.5, -0.5], [-0.5, -0.5, 0.5]]}]
 
         Returns:
             A list of dictionaries, one for each particle type in the system.
         """
-        return super(convex_spheropolyhedron, self)._return_type_shapes()
+        return super(ConvexSpheropolyhedron, self)._return_type_shapes()
 
 
-class ellipsoid(_HPMCIntegrator):
+class Ellipsoid(_HPMCIntegrator):
     R""" HPMC integration for ellipsoids (2D/3D).
 
     Args:
         seed (int): Random number seed.
-        d (float): Maximum move displacement, Scalar to set for all types, or a dict containing {type:size} to set by type.
-        a (float): Maximum rotation move, Scalar to set for all types, or a dict containing {type:size} to set by type.
+        d (float): Maximum move displacement, Scalar to set for all types, or a
+          dict containing {type:size} to set by type.
+        a (float): Maximum rotation move, Scalar to set for all types, or a dict
+          containing {type:size} to set by type.
         move_ratio (float): Ratio of translation moves to rotation moves.
         nselect (int): The number of trial moves to perform in each cell.
-        restore_state(bool): Restore internal state from initialization file when True. See :py:class:`_HPMCIntegrator`
-                             for a description of what state data restored. (added in version 2.2)
+        deterministic (bool): Make HPMC integration deterministic on the GPU
 
     Ellipsoid parameters:
 
-    * *a* (**required**) - principle axis a of the ellipsoid (radius in the x direction) (distance units)
-    * *b* (**required**) - principle axis b of the ellipsoid (radius in the y direction) (distance units)
-    * *c* (**required**) - principle axis c of the ellipsoid (radius in the z direction) (distance units)
-    * *ignore_statistics* (**default: False**) - set to True to disable ignore for statistics tracking
+    * *a* (**required**) - principle axis a of the ellipsoid (radius in the x
+      direction) (distance units)
+    * *b* (**required**) - principle axis b of the ellipsoid (radius in the y
+      direction) (distance units)
+    * *c* (**required**) - principle axis c of the ellipsoid (radi
```

Comment:
We don't need to specify for all particles. The default is the default.

Reply:
I am confused what you mean by this and the above comment. Should I remove the '(distance units)'?